### PR TITLE
CI: add better integration on GitHub Actions.

### PR DESCRIPTION
**Pull request by eloycoto**
This change adds a way to test the following things in GitHub actions:

- Build using maven build as it was.
- A new job to report jacoco reports.
- A super simple way to deploy things to kubernetes for example.
- A simple way to run the example script.
- A big refactor on example script with colors and assertions.
- Added cache on maven when build

The workflow is like this one:
https://github.com/redhat-developer/parodos/actions/runs/4232739123

### CODE REVIEW

File: hack/manifests/base/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+resources:
```
Comment (pkliczewski) [inline hack/manifests/base/kustomization.yaml:1]: Do you think it makes sense to have more manifests? We have local profile but we need integration and production as well.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+
+COMPOSE ?= docker-compose
+DOCKER ?= docker
+
+ORG=quay.io/parados/
+WORKFLOW_SERVICE_IMAGE=workflow-service
+NOTIFICATION_SERVICE_IMAGE=notification-service
+
+GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD | sed s,^EloyGH$$,test,g)
```
Comment (pkliczewski) [inline Makefile:9]: Do we need `EloyGH` here?

File: hack/manifests/base/kustomization.yaml (hunk 1 above)
Comment (eloycoto) [reply hack/manifests/base/kustomization.yaml:1]: At the moment I added the testing folder, and from there we can evolve in case that it's needed ;-)

Comment (pkliczewski) [reply hack/manifests/base/kustomization.yaml:1]: OK, make sense

**Review by lshannon — APPROVED** [review]

**Pull request merged**